services/dedup/badger/badger.go

```diff
@@ -66,13 +109,13 @@ func (d *badgerDB) Set(kvs []KeyValue) error {
 	return txn.Commit()
 }
 
-func (d *badgerDB) Close() {
+func (d *BadgerDB) Close() {
 	close(d.close)
 	<-d.gcDone
 	_ = d.badgerDB.Close()
 }
 
-func (d *badgerDB) start() {
+func (d *BadgerDB) Start() {
```

Comment:
I would try to "hide" Start method, we can implicitly call start once the first time a call to Get or Set is called. 

We should also return an error instead of a panic.  

```go
func (d *BadgerDB) Set(kvs []types.KeyValue) error {
    err := d.init() 
    if err != nil {
        return err
    }
}
```
```suggestion
func (d *BadgerDB) init() error {
    d.once.Do(func () {
      d.badgerDB, d.initErr = badger.Open(d.opts)
      ....
    })
    return d.initErr
```

Reply:
We would also need to change the signature for Get

```
func (d *BadgerDB) Get(key string) (int64, error)
```

This will be useful for scyllaDB client, as any method call could result into an error